discover/portal/articles/210-setting-up/01-system-settings/02-config-files/02-factory-configurations.markdown

```diff
@@ -0,0 +1,101 @@
+# Factory Configurations [](id=factory-configurations)
+
+Configurations supporting multiple independent occurrences of the same
+configuration entry are called *factory configurations*. 
+
++$$$
+
+**Factory Configuration Example:** 
+[JAX-WS and JAX-RS web services](/develop/tutorials/-/knowledge_base/7-1/jax-ws-and-jax-rs) 
+are supported. These services must use a [CXF
+Endpoint](/develop/tutorials/-/knowledge_base/7-1/jax-ws-and-jax-rs#cxf-endpoints),
+which is a context path where the web services are deployed and accessed.
+Endpoints can be created via factory configuration by navigating to the CXF
+Endpoints entry in System Settings (System Settings &rarr; Platform &rarr; Web
+API &rarr; CXF Endpoints). Click *ADD*, enter the desired configuration values,
+then repeat the process to add as many CXF Endpoint configurations as needed.
+Creating CXF Endpoint configurations also creates CXF Endpoints themselves. This
+is how factory configurations work. 
+
+$$$
+
+If a service is meant to support factory configurations, its System Settings
+entry has an ADD button. 
+
+![Figure 1: If a System Settings entry has an ADD button, it's suitable for factory configurations.](../../../../images/factory-configuration-entry.png)
+
+As with single-instance configurations, you can set factory configurations in 
+the System Settings interface (as described in the example above) or via 
+configuration files. Name a standard single-instance configuration file like
+this: 
+
+    my.service.ServiceConfiguration.config
+
+If your service supports factory configurations, use the convention of calling
+the configuration's first instance `-default.config`: 
+
+    my.service.ServiceConfiguration-default.config
+
+The next instance contains a unique *subname* (something other than *default*).
+It's good practice to use a descriptive name: 
+
+    my.service.ServiceConfiguration-port9080.config
+
+To follow the CXF Endpoints example described above, if Liferay's developers had
+shipped an initial CXF Endpoint `.config` file with @product@, it would have 
+been named: 
+
+    com.liferay.portal.remote.cxf.common.configuration.CXFEndpointPublisherConfiguration-default.config
+
+If this `-default.config` configuration specifies a context path for *REST* web 
+services, and you create another endpoint with a different context path for 
+*SOAP* web services, your second configuration file could be named: 
+
+    com.liferay.portal.remote.cxf.common.configuration.CXFEndpointPublisherConfiguration-soap.config
+
++$$$
+
+**Note:** Some System Settings entries (like the CXF Endpoints entry) don't ship
+with a configuration file, so anything you create is the first occurrence. 
+However, if you configure one and export it to obtain the `.config` file, it's 
+not named `-default.config`. Instead it's given a guaranteed unique identifier
```

Comment:
Reading this made me think that -default had some special meaning. It does not, it's just a convention we use. That suffix never has any special meaning.

Reply:
Edited:
>**Note:** Some System Settings entries (like the CXF Endpoints entry) don't ship
>with a configuration file, so anything you create is the first occurrence.
>However, if you configure one and export it to obtain the `.config` file, it
>doesn't use the `-default.config` naming convention. Instead it's given a
>guaranteed unique identifier for its subname, like this: